### PR TITLE
Switch CoreDNS Deployment to (node-local) DaemonSet

### PR DESCRIPTION
**Pull request by linki**
Replaces the node-local DNS setup via `dnsmasq` with dedicated and independent CoreDNS instances per node.

Client-side DNS configuration is the same:
* First DNS server is the local node's IP which is backed by a single CoreDNS Pod on the same node
* Second DNS server is a `ServiceIP` which is backed by a set of CoreDNS Pods across the cluster
  * Note, that in this PR we're using the very same DaemonSet's Pods as targets (this could be changed to a dedicated fallback CoreDNS as a Deployment as before)

### CODE REVIEW

File: cluster/manifests/coredns/configmap.yaml [hunk 1]
```diff
@@ -9,7 +9,7 @@ data:
   Corefile: |
     .:53 {
         errors
-        health
+        health :9154
```
Comment (linki) [inline cluster/manifests/coredns/configmap.yaml]: This needed as the default `8080` in combination with `hostNetwork` clashes.

File: cluster/manifests/deletions.yaml [hunk 2]
```diff
@@ -8,4 +8,16 @@ pre_apply:
 #pre_apply: []
 
 # everything defined under here will be deleted after applying the manifests
-post_apply: []
+post_apply:
```
Comment (szuecs) [inline cluster/manifests/deletions.yaml]: We don't want to delete this in one step. We want to have these running on old nodes as discussed on the whiteboard.

**Comment by linki** [comment]
rollout plan:

- [x] [rollout 1](https://github.com/zalando-incubator/kubernetes-on-aws/pull/1584):
* add `instance=cluster-dns` to CoreDNS Deployment PodSpec
* update coredns config to final version (health check port change)

- [x] [rollout 2](https://github.com/zalando-incubator/kubernetes-on-aws/pull/1573):
* add node labels `cluster-dns=dnsmasq` to all nodes via userdata (rolls all nodes)
  * rolling could be avoided by running a script over all clusters and nodes labelling them.
* add `instance=cluster-dns` to CoreDNS Deployment Selector and Service Matcher => no change
* add CoreDNS DS (with `instance=node-dns`) with nodeSelector `cluster-dns!=dnsmasq` => won't run
* update dnsmasq DS with nodeSelector `cluster-dns=dnsmasq` => keeps running

- [ ] [rollout 3](https://github.com/zalando-incubator/kubernetes-on-aws/pull/1603):
* remove node labels `cluster-dns=dnsmasq` via userdata (rolls all nodes to their final version)
  * new nodes will run CoreDNS DS pods, dnsmasq DS gets phased out
  * in case of labelling manually in rollout 2 this becomes a manually controlled force roll of all nodes

- [ ] rollout 4:
* remove dnsmasq DS => no change
* remove CoreDNS Deployment => deletes fallback DNS server (check again)
* remove `instance=node-dns` from CoreDNS DS and Selector (no change, merely for final version)

/cc @njuettner @aermakov-zalando @szuecs

**Comment by linki** [comment]
merged together to alpha with https://github.com/zalando-incubator/kubernetes-on-aws/pull/1580

File: cluster/manifests/coredns/deployment-coredns.yaml [hunk 3]
```diff
@@ -1,4 +1,4 @@
-apiVersion: apps/v1
+apiVersion: extensions/v1beta1
```
Comment (linki) [inline cluster/manifests/coredns/deployment-coredns.yaml:1]: Needed to make `kubectl apply` work for this PR. Can be switched back later.

**Comment by linki** [comment]
lgtm, should be merged to `alpha` together with other userdata changes.

**Comment by njuettner** [comment]
👍

**Comment by linki** [comment]
:+1:

**Pull request merged**